actions/release-candidate/src/main.js

```diff
@@ -26,46 +26,97 @@ async function run() {
 		const tlp_dir = core.getInput("tlp_dir", { required: true });
 		const project_id = core.getInput("project_id", { required: true });
 		const project_dir = core.getInput("project_dir");
-		const gpg_signing_key = core.getInput("gpg_signing_key", { required: true });
-		const svn_username = core.getInput("svn_username", { required: true });
-		const svn_password = core.getInput("svn_password", { required: true });
-		const nexus_username = core.getInput("nexus_username", { required: true });
-		const nexus_password = core.getInput("nexus_password", { required: true });
-		let publish = core.getBooleanInput("publish");
+		const publish = core.getBooleanInput("publish");
+		let gpg_signing_key = "";
+		let gpg_signing_public_key = "";
+		let gpg_signing_key_id = "";
+		const gitTagPrefix = "refs/tags/";
+		// get the actual project version, this requires a 'VERSION' file at
+		// the root of the repository
+		const project_version = fs.readFileSync("VERSION").toString().trim();
+		const is_snapshot = project_version.includes("-SNAPSHOT");
+		const do_publish =
+			// Note that publishing could be disabled if the publish input was explicitly set
+			// to false
+			publish
+			// enable publishing for tags
+			&& (github.context.eventName == "push" && github.context.ref.startsWith(gitTagPrefix))
+			// enable publishing for non-snapshot builds and ASF builds.
+			&& (!is_snapshot && process.env.GITHUB_REPOSITORY_OWNER == "apache")
+
+		if (do_publish) {
+			gpg_signing_key = core.getInput("gpg_signing_key", {required: true});
+		}
+
+		if (gpg_signing_key.trim() === "") {
+			// Generate keypair (non-interactive)
+			await exec("gpg", ["--batch", "--yes", "--passphrase", '', "--quick-generate-key", process.env.USER ]);
```

Comment:
Generating a secret key automatically imports it, so it isn't necessary to list the secret/public key and to import it later. I think this can just be

```ts
if (gpg_signing_key.trim() === "") {
   // Generate keypair (non-interactive)
  await exec("gpg", ["--batch", "--yes", "--passphrase", '', "--quick-generate-key"]);
} else {
  await exec("gpg", ["--batch", "--import", "--import-options", "import-show"])
}
```
I'm not sure if --quick-generate-key will output the key id, so maybe you can do tweak and do the --list-secret-keys command after this to extract the key id out of that. That would probably be a more reliable way to get the key id anyways instead of our current thing of just searching trying to match 40 characters.